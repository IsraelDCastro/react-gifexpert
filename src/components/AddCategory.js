import React, { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState('');
  const handleInputChange = (input) => {
    setInputValue(input.target.value)
  }
  const handleSubmit = (form) => {
    form.preventDefault();
    if ( inputValue.trim().length > 2) {
      setCategories( (categories) => [inputValue, ...categories] );
      setInputValue('');
    }
  }

  return (
    <form onSubmit={handleSubmit}>
      <input
        type="text"
        value={inputValue}
        onChange={ handleInputChange }
      />
    </form>
  )
}

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired
}


