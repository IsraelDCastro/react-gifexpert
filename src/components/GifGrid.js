import React from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";
import PropTypes from "prop-types";

export const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifs(category);

  return (
    <>
      <h4 className="animate__animated animate__fadeIn">{ category }</h4>
      { loading && <p className="animate__animated animate__flash">"Loading..."</p> }
      <div className="card-grid">
        { images.map((image) => (
          <GifGridItem key={image.id} {...image} />
        ))}
      </div>
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}
