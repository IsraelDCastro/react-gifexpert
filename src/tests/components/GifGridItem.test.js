import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

describe("Tests for <GifGridItem />", () => {
  const title = "A Title";
  const url = "https://localhost/something.jpg";
  const wrapper = shallow(
    <GifGridItem
      title={title}
      url={url}
    />
  );

  test("must be render <GifGridItem /> correctly", () => {

    expect(wrapper).toMatchSnapshot();
  });

  test("must have a paragraph with title", () => {
    const p = wrapper.find('p');

    expect(p.text().trim()).toBe(title);
  });

  test("must have an image and alt of props", () => {
    const img = wrapper.find('img');

    expect(img.prop('src')).toBe(url);
    expect(img.prop('alt')).toBe(title);
  });

  test("must have class animate__fadeIn", () => {
    const div = wrapper.find('div');
    const className = div.prop('className');
    expect(className).toMatch("animate__fadeIn");
    expect(className.includes('animate__fadeIn')).toBe(true);
  });
});
