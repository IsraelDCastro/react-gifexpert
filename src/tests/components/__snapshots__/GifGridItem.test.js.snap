// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <GifGridItem /> must be render <GifGridItem /> correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="A Title"
    src="https://localhost/something.jpg"
  />
  <p>
    A Title
  </p>
</div>
`;
